src/EndpointsControllers/booksController.js

```diff
@@ -0,0 +1,183 @@
+import { connection, getUserBySession, recordExist } from '../Helpers/index.js';
+import { MAX_BOOKS_FOR_READER } from '../Helpers/constants.js';
+import { DefaultController } from './defaultController.js';
+import sql from 'mysql-bricks';
+import { dbTablesNames } from '../enums/index.js';
+
+class BooksController extends DefaultController {
+    constructor () {
+        const updatableFields = ['title', 'author', 'amount'];
+        const searchableFields = ['title', 'author', 'departmentID'];
+
+        super(dbTablesNames.BOOKS, updatableFields, searchableFields);
+    }
+
+    add () {
+        return async (req, res, next) => {
+            try {
+                if (!(await recordExist(req.body.departmentID, dbTablesNames.DEPARTMENTS))) {
+                    return res.status(400).send(`Department with ID ${req.body.departmentID} doesn't exist`);
+                }
+
+                const fields = { title: true, author: true, departmentID: true, amount: true };
+                const values = this._getValuesFromRequestBody(req.body, fields);
+
+                if (typeof values === 'string') {
+                    return res.status(404).send(`Field with name '${values}' is necessary`);
+                }
+
+                const query = sql
+                    .insert(this._tableName, Object.keys(fields))
+                    .values(values)
+                    .toParams({ placeholder: '?' });
+
+                await connection.query(query.text, query.values);
+
+                res.status(200).send('ok');
+            } catch (e) {
+                next(e);
+            }
+        };
+    }
+
+    get () {
+        return async (req, res, next) => {
+            try {
+                await super.get(res, req);
+            } catch (e) {
+                next(e);
+            }
+        };
+    }
+
+    givenAmount () {
+        return async (req, res, next) => {
+            try {
+                const id = req.body.id;
+
+                const query = sql
+                    .select(sql('COUNT(id) as givenAmount'))
+                    .from('givenbooks')
+                    .where(sql.eq('bookID', id))
+                    .toParams({ placeholder: '?' });
+
+                const givenAmount = (await connection.query(query.text, query.values))[0][0];
+
+                res.status(200).json(givenAmount);
+            } catch (e) {
+                next(e);
+            }
+        };
+    }
+
+    update () {
+        return async (req, res, next) => {
+            try {
+                if (req.amount < 0) {
+                    return res.status(400).send('Amount must be great than 0');
+                }
+
+                await super.update()(req, res);
```

Comment:
I don't see that `update` function returns another function to use double parentheses here... Is it supposed to be `super.update(req, res)`?